pcsx2/MemoryCardFolder.cpp

```diff
@@ -41,17 +44,39 @@ const ryml::Tree treeFromString(const std::string& s)
 // A helper function to parse the YAML file
 std::optional<ryml::Tree> loadYamlFile(const wxString& filePath)
 {
-	auto path = Path::FromWxString(filePath);
-	if (!fs::exists(path))
+	ryml::Callbacks rymlCallbacks = ryml::get_callbacks();
+	rymlCallbacks.m_error = [](const char* msg, size_t msg_len, ryml::Location loc, void*) {
+		throw std::runtime_error(fmt::format("[YAML] Parsing error at {}:{} (bufpos={}): {}",
+			loc.line, loc.col, loc.offset, msg));
+	};
+	ryml::set_callbacks(rymlCallbacks);
+	c4::set_error_callback([](const char* msg, size_t msg_size) {
+		throw std::runtime_error(fmt::format("[YAML] Internal Parsing error: {}",
+			msg));
+	});
+	try
 	{
-		return std::nullopt;
-	}
+		auto path = Path::FromWxString(filePath);
```

Comment:
All of this can be replaced by `FileSystem::ReadFileToString()` (and remove the wxString dependency as well).

Reply:
imo I would save that for when this file is properly cleaned of wx types, there are over a hundred wx dependencies remaining in this file.

Reply:
But why introduce more? It's just adding more work for later.

Reply:
1 out of over 125 isn't really a big deal, with 125 or 126 the person who converts this file is going to hate themselves :P

Reply:
Right, but with all respect, this is written pretty poorly. Redundant reads/copies, when there's a function which does all the work in our wrapper namespace (~10 lines down to 1). It's pretty similar to the bit I replaced in gamedb loading.

Literally just
```
std::optional<std::string> buffer = FileSystem::ReadFileToString(StringUtil::wxStringToUTF8String(path).c_str());
if (!buffer.has_value())
  return std::nullopt;
const ryml::Tree tree = treeFromString(buffer->c_str());
```

I know the old code was written this way, but given it's being modified anyway, it makes sense to switch it over and save time later on.

Reply:
Then I would recommend we make it work by massaging it for now, then somebody will need to rewrite it to make it not suck. Maybe Silent would like to look at it since he added all the index stuff.

Reply:
Suggestion committed, thanks.